samples/extensions/extended_dynamic_state2/extended_dynamic_state2.cpp

```diff
@@ -94,17 +94,34 @@ void ExtendedDynamicState2::load_assets()
 {
 	load_scene("scenes/primitives/primitives.gltf");
 
-	std::vector<SceneNode> scene_elements;
-	// Store all scene nodes in a linear vector for easier access
-	for (auto &mesh : scene->get_components<vkb::sg::Mesh>())
+	std::vector<SceneNode>       scene_elements;
+	std::vector<vkb::sg::Node *> node_scene_list = scene->get_root_node().get_children();
+	vkb::sg::Node *              node            = nullptr;
+
+	for (size_t list_it = 0; node_scene_list.size() > list_it; list_it++)
 	{
-		for (auto &node : mesh->get_nodes())
+		node = node_scene_list.at(list_it);
```

Comment:
In PR https://github.com/KhronosGroup/Vulkan-Samples/pull/611/files recently opened Andreas is removing the usage of "at(i)" method (in favour of [] operator).
He is adding asserts if needed to check boundaries, but since it is for loop we do not need one.

Change proposal:
for (size_t list_it = 0; list_it < node_scene_list.size(); ++list_it)
{
	node = node_scene_list[list_it];

Reply:
The changes have been implemented